openproblems/tasks/_batch_integration/batch_integration_graph/methods/baseline.py

```diff
@@ -0,0 +1,102 @@
+from .....tools.decorators import method
+from .....tools.utils import check_version
+
+import numpy as np
+import scipy.sparse
+
+
+def _set_uns(adata):
+    adata.uns["neighbors"] = adata.uns["uni"]
+    adata.uns["neighbors"]["connectivities_key"] = "connectivities"
+    adata.uns["neighbors"]["distances_key"] = "distances"
+
+
+@method(
+    method_name="No Integration",
+    paper_name="No Integration (baseline)",
+    paper_url="https://openproblems.bio",
+    paper_year=2022,
+    code_url="https://github.com/openproblems-bio/openproblems",
+    is_baseline=True,
+)
+def no_integration(adata, test=False):
+    adata.obsp["connectivities"] = adata.obsp["uni_connectivities"]
+    adata.obsp["distances"] = adata.obsp["uni_distances"]
+    _set_uns(adata)
+    adata.uns["method_code_version"] = check_version("openproblems")
+    return adata
+
+
+def _randomize_subgraph(distances, connectivities):
+    row = np.random.choice(
+        np.arange(distances.shape[0]), len(distances.data), replace=True
+    )
+    col = np.random.choice(
+        np.arange(distances.shape[0]), len(distances.data), replace=True
+    )
+    while np.any(np.any(row == col)):
+        # eliminate self-loops
+        row[row == col] = np.random.choice(
+            np.arange(distances.shape[0]), np.sum(row == col)
+        )
+    # eliminate duplicates
+    row, col = zip(*set(zip(row, col)))
+    distances_data = np.random.permutation(distances.data)[: len(row)]
+    distances_out = scipy.sparse.coo_matrix(
+        (distances_data, (row, col)), shape=distances.shape
+    ).tocsr()
+    connectivities_data = np.random.permutation(connectivities.data)[: len(row)]
+    connectivities_out = scipy.sparse.coo_matrix(
+        (connectivities_data, (row, col)), shape=connectivities.shape
+    ).tocsr()
+    return distances_out, connectivities_out
```

Comment:
You're right that removing duplicates marginally reduces total number of edges (by a factor of 1/n). This should be a pretty small effect, but I'm happy to address it if you think it necessary.

> Why not just randomize the labels over the existing connectivity structure?

I thought about this, but decided against it because randomising labels keeps the existing graph structure, e.g. if the original graph is highly cliquey (many tight clusters) then the randomized graph will also have many tight clusters, where this function instead gives a graph with truly random connections within each partition.